azurelinuxagent/ga/update.py

```diff
@@ -730,6 +739,21 @@ def _write_pid_file(self):
 
         return pid_files, pid_file
 
+    def _send_heartbeat_telemetry(self, protocol):
+        if self.last_telemetry_heartbeat is None:
+            self.last_telemetry_heartbeat = datetime.utcnow() - UpdateHandler.TELEMETRY_HEARTBEAT_PERIOD
+
+        if datetime.utcnow() >= (self.last_telemetry_heartbeat + UpdateHandler.TELEMETRY_HEARTBEAT_PERIOD):
+            dropped_packets = self.osutil.get_firewall_dropped_packets(protocol.get_endpoint())
+            msg = "{0};{1};{2}".format(self.heartbeat_counter, self.heartbeat_id, dropped_packets)
+
+            add_event(name=AGENT_NAME, version=CURRENT_VERSION, op=WALAEventOperation.HeartBeat, is_success=True,
+                      message=msg, log_event=False)
+            self.heartbeat_counter += 1
+
+            logger.info(u"Agent {0} is running as the goal state agent", CURRENT_AGENT)
```

Comment:
Suggestion: adding `[PERIODIC]` to this message would make it clearer that it's just a diagnostic message, not that the agent was just started (which I think might end up being an outsider's interpretation).

Reply:
Good idea. But I'll add `[HEARTBEAT]` rather than the Periodic tag to signify that its heartbeat, because we do use Periodic for other things.